### ISSUE TITLE
Native batch normalization operator

### ISSUE THREAD

**Issue by aclex** [kendryte/nncase#146]
When debugging batch normalization operator import for ONNX I've noticed there's some traces of batchnorm support in K210 part of low-level IR and code generation parts. So what I'm thinking of is if maybe there's serious reason that there's no native batch normalization operator in `hlir`. Is it just not yet implemented or there're some serious problems with it? Obviously, there's no such signs of its support in CPU `llir`, but that's obviously not that hard, while it probably could seriously speed up the batchnorm compared to the emulated version, which presents a bunch of arithmetic operators (though quite simple ones) to implement exactly the same.

**Comment by sunnycase** [comment]
I think for inference a bn can be imported as `mul` and `add`.

**Comment by aclex** [comment]
@sunnycase that's indeed right, but seems to add much noise to the whole model structure when imported.
I've managed to import successfully Mobilenetv3-small + SSDLite model taken from my experiments [here](https://github.com/aclex/detection-experiments) using ONNX importer backend and I'm experimenting with making it work now. For now the final model is about 10x larger than stock face recognition TinyYOLO memory-wise and also have certain convolution diverged during optimization, which is why they are left in float. So I'm trying to figure out the problem with it and what have attracted my attention is quite a big snippets per each batchnorm. I have a couple of ideas how to optimize them while importing, but noticed there's some hardware support in K210, hence the question.

**Comment by sunnycase** [comment]
Generally bn after conv2d or matmul can be fused into their's weights.
Maybe run a onnx optimization can help. There also should be a bn fusion pass in nncase transforms in the future.

**Comment by aclex** [comment]
@sunnycase Sounds like nice optimization. Do you think it's worth to be performed as early as during the model import or rather better off staying in the model graph optimization stage?

Another little question here: is there any similar constant nodes collapsing strategy during the model optimization now? I mean, there are some constant-demanding operators like Relu6 which overflows the graph with many constant nodes with the same constant value (`1` or `6` or `0`). I'm trying to de-duplicate them during the import in my experiments, without big success for now though, but maybe I'm doing the redundant job and it's already performed later during the final model optimization? Or maybe such operators are generally hinder the overall performance because of this and not really recommended to use (I then can substitute to e.g. ordinary ReLU)?

**Comment by sunnycase** [comment]
1.x will lower bn in import phase. 2.x in development will have native bn op.
